src/modules/fetcher/FetcherModule.js

```diff
@@ -18,18 +18,24 @@ class FetcherModule {
     progress: ProgressCallbackType
   ): Promise<FetchResultType> {
     if (FetcherModule.currentlyFetching) {
-      return Promise.reject(new Error('Already fetching!'))
+      throw new Error('Already fetching!')
     }
     if (isEmpty(targetFilePaths)) {
-      return Promise.resolve({})
+      return {}
     }
     FetcherModule.currentlyFetching = true
 
     const subscriptions = []
     subscriptions.push(NativeFetcherModuleEmitter.addListener('progress', progress))
 
     try {
-      return await NativeFetcherModule.fetchAsync(targetFilePaths)
+      const result = await NativeFetcherModule.fetchAsync(targetFilePaths)
+
```

Comment:
Add a comment of why this could happen (or that we don't know why this might happen)

Reply:
Yeah I dont know :D Will add a comment